src/garage/sampler/ray_batched_sampler.py

```diff
@@ -0,0 +1,216 @@
+"""This is an implementation of an on policy batch sampler.
+
+Uses a data parallel design.
+Included is a sampler that deploys sampler workers.
+
+The sampler workers must implement some type of set agent parameters
+function, and a rollout function
+"""
+from collections import defaultdict
+import pickle
+
+import numpy as np
+import psutil
+import ray
+
+from garage.experiment import deterministic
+from garage.misc import tensor_utils
+from garage.misc.prog_bar_counter import ProgBarCounter
+from garage.sampler.base import BaseSampler
+
+
+class RaySampler(BaseSampler):
+    """Collects Policy Rollouts in a data parallel fashion.
+
+    Args:
+        - algo: A garage algo object
+        - env: A gym/akro env object
+        - should_render(bool): should the sampler render the trajectories
+        - sampler_worker_cls: If none, uses the default SamplerWorker
+            class
+
+    """
+
+    def __init__(self,
+                 algo,
+                 env,
+                 seed,
+                 should_render=False,
```

Comment:
sampler workers should not render ever.

Reply:
@krzentner said he might need the option to render a worker, if I understood him correctly.

Reply:
It's certainly rare to need to render workers, but there's vaguely two reasons it seems convenient to me:
1. If you're debugging some behavior that you only see in the sampler, being able to render some of the environments can be a really useful debugging tool.
2. It's sometimes easier to make one of the workers render than to start a totally separate program to do rendering.